File: android/src/main/java/com/bluetrum/devicemanager/cmd/deviceinfo/DeviceInfoByteCallable.java
package com.bluetrum.devicemanager.cmd.deviceinfo;

import androidx.annotation.NonNull;

/**
 * @deprecated
 * Use class inheriting from {@link com.bluetrum.devicemanager.cmd.payloadhandler.PayloadHandler}.
 */
@Deprecated
public final class DeviceInfoByteCallable extends DeviceInfoCallable<Byte> {

    public DeviceInfoByteCallable(@NonNull final byte[] payload) {
        super(payload);
    }

    @Override
    public Byte call() throws Exception {
        final byte[] payload = getPayload();
        if (payload.length == 1) {
            return payload[0];
        }
        return null;
    }

}

File: ios/Classes/DeviceManager/Command/DeviceInfo/DeviceInfoByteCallable.swift
//
//  DeviceInfoByteCallable.swift
//  DeviceManager
//
//  Created by Bluetrum.
//  

import Foundation

@available(*, deprecated, message: "Use class inheriting from 'PayloadHandler'.")
public final class DeviceInfoByteCallable: AbstractDeviceInfoCallable {
    
    public override func callAsFunction() -> AnyObject? {
        if payload.count == 1 {
            return payload[0] as AnyObject
        }
        return nil
    }
    
}
